sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/ExpressionInfo.java

```diff
@@ -40,8 +40,11 @@
     private String deprecated;
 
     private static final Set<String> validGroups =
-        new HashSet<>(Arrays.asList("agg_funcs", "array_funcs", "datetime_funcs",
-            "json_funcs", "map_funcs", "window_funcs"));
+        new HashSet<>(Arrays.asList("agg_funcs", "array_funcs", "binary_funcs", "bitwise_funcs",
+            "comparison_funcs", "conditional_funcs", "conversion_funcs", "csv_funcs",
+            "datetime_funcs", "generator_funcs", "grouping_funcs", "json_funcs", "logical_funcs",
+            "map_funcs", "math_funcs", "misc_funcs", "regex_funcs", "string_funcs", "struct_funcs",
+            "window_funcs", "xml_funcs"));
```

Comment:
I tried to follow the way [Presto](https://prestodb.io/docs/current/functions.html) has classified its functions, but all this is up for a change.

Reply:
I was also considering making these values constants:
```    
    String GROUP_AGG_FUNCS = "agg_funcs";
    String GROUP_ARRAY_FUNCS = "array_funcs";
    String GROUP_BINARY_FUNCS = "binary_funcs";
...
```
Perhaps in the `ExpressionDescription` interface, but I'm not 100% sure if its better.

Reply:
hm, I see. Adding the constraints look okay because they make it easier to track these groups on IDEs, I think.